### ISSUE TITLE
Add Swagger syntax for configuring LROs

### ISSUE THREAD

**Issue by xirzec** [Azure/autorest.typescript#1136]
Specifically things like tying together two separate paths for creating and polling an LRO and any other non-inferable configuration.

**Comment by qiaozha** [comment]
Is this a requirement for swagger team instead of code generator?

**Comment by qiaozha** [comment]
@deyaaeldeen I wonder why you need the polling url to be pre-defined in swagger from codegen side ?  I agree that if we have them pre-defined in swagger, we can add some validation on them to make sure lro really works. but for the client side, it still needs to get the value from long running operation final-state-via header, right? 
cc @ramya-rao-a

**Comment by deyaaeldeen** [comment]
@leni-msft yes I am familiar with the extension and it does not specify all the information needed (e.g., when to stop polling in dataplane services LROs).

@qiaozha we actually need the path to poll, a spec for when to stop polling, a list of places that need to be updated after each poll, and the location of the result. Otherwise, I will have to determine all these information either dynamically based on inspecting the response structure (like what we do for mgmt-plane pkgs) or by manually-written convenience-layer code (like what we do for data-plane pkgs).

**Comment by qiaozha** [comment]
Get your point.

**Comment by leni-msft** [comment]
@deyaaeldeen In this case, I'd suggest to submit PR for your extension proposal, e.g., https://github.com/Azure/autorest/pull/4150
